### ISSUE TITLE
Mitigate CVE-2016-1000111 ("httpoxy")

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8623]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8623|
|Type|defect|
|Created|2016-07-18 19:16:39Z|
|Branch|https://github.com/twisted/twisted/tree/8623-httpoxy-hilarity-2|

As dumb as this CVE is, we should remove the vuln.

<details><summary>Searchable metadata</summary>

```
trac-id__8623 8623
type__defect defect
reporter__hawkowl hawkowl
priority__highest highest
milestone__Twisted_16_3_1 Twisted 16.3.1
branch__8623_httpoxy_hilarity_2 8623-httpoxy-hilarity-2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__None None
time__1468869399817220 1468869399817220
changetime__1470834131839548 1470834131839548
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

PR open at https://github.com/twisted/twisted/pull/482 .

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_5"></a>|@Lukasa set owner to @hawkowl|
|-|-|

LGTM, merge when the builds all go green (if they all go green).

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset 3853ab613d17983638b6039e0878104f9ee1ba92
```
#!CommitTicketReference repository="" revision="3853ab613d17983638b6039e0878104f9ee1ba92"
Merge 8623-httpoxy-hilarity-2: Mitigate CVE-2016-1000111

Author: hawkowl
Reviewer: lukasa
Fixes: #8623
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

In changeset 5df208a1b59168932d64b399fd7aa605f5554135
```
#!CommitTicketReference repository="" revision="5df208a1b59168932d64b399fd7aa605f5554135"
Merge 8623-httpoxy-hilarity-2: Mitigate CVE-2016-1000111

Author: hawkowl
Reviewer: lukasa
Fixes: #8623
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl commented|
|-|-|

In changeset 69707bb1aa55b3a6cec5e02df01d34d2a93c2519
```
#!CommitTicketReference repository="" revision="69707bb1aa55b3a6cec5e02df01d34d2a93c2519"
Merge 8623-httpoxy-hilarity-2: Mitigate CVE-2016-1000111

Author: hawkowl
Reviewer: lukasa
Fixes: #8623
```
